tests/integration/test_replication_credentials/test.py

```diff
@@ -127,6 +127,54 @@ def test_credentials_and_no_credentials(credentials_and_no_credentials_cluster):
     node8.query("insert into test_table values ('2017-06-22', 222, 1)")
     time.sleep(1)
 
-    assert node7.query("SELECT id FROM test_table order by id") == '111\n'
+    assert node7.query("SELECT id FROM test_table order by id") == '111\n222\n'
```

Comment:
But why it now contains the newly inserted record? AFAICS it still should fail due to authentication error

Reply:
Node7 has no credentials configured so it accepts everything. This is useful to migrate from no credentials to configured credentials (unauthenticated server should allow authenticated client until server is reconfigured).. configuring a credential instead in null would retain existing behavior

Reply:
But it is node8 who do not have credentials, am I missing something?

Reply:
Node7 can replicate from node8; node8 can't replicate from node7. Null credentials allow any connection. This change in behavior is documented in message.

Reply:
Previously if the client sent credentials but auth want configured on the server it would reject. With this change ba server w/o auth configured accepts any credential